pytorch_lightning/core/lightning.py

```diff
@@ -1207,12 +1207,18 @@ def training_step(...):
                 self.manual_backward(loss, opt_a)
                 opt_a.step()
         """
+        if optimizer is not None:
+            rank_zero_warn(
+                "`optimizer` argument to `manual_backward` is deprecated in v1.2 and will be removed in v1.4",
+                DeprecationWarning
+            )
+
         # make sure we're using manual opt
         self._verify_is_manual_optimization('manual_backward')
 
         # backward
         self._running_manual_backward = True
-        self.trainer.train_loop.backward(loss, optimizer, -1, *args, **kwargs)
+        self.trainer.train_loop.backward(loss, None, None, *args, **kwargs)
```

Comment:
This looks really confusing, I'd prefer if we were explicit but I've seen people be opinionated here:

```suggestion
        self.trainer.train_loop.backward(
            result=loss,
            optimizer=None,
            opt_idx=None,
            *args, **kwargs
        )
```

Reply:
You are right !